scripts/solr_builder/solr_builder/solr_builder_main.py

```diff
@@ -130,30 +136,42 @@ def query_batched(self, query, size, cursor_name=None, cache_json=False):
         cur.close()
 
     @staticmethod
-    def _get_lite_metadata(ocaids, rows=1000):
+    def _get_lite_metadata(ocaids, _recur_depth=0, _max_recur_depth=3):
```

Comment:
it would be great to have a comment here explaining the recursive divide and conquer strategy as someone seeing this code for the first time is probably going to say TL;DR